onsset/onsset.py

```diff
@@ -610,114 +663,29 @@ def grid_penalties(self):
-                return 2
-            else:
-                return 1
-
-        def classify_substation_dist(row):
-            substation_dist = row[SET_SUBSTATION_DIST]
-            if substation_dist <= 0.5:
-                return 5
-            elif substation_dist <= 1:
-                return 4
-            elif substation_dist <= 5:
-                return 3
-            elif substation_dist <= 10:
-                return 2
-            else:
-                return 1
-
-        def classify_land_cover(row):
-            land_cover = row[SET_LAND_COVER]
-            if land_cover == 0:
-                return 1
-            elif land_cover == 1:
-                return 3
-            elif land_cover == 2:
-                return 4
-            elif land_cover == 3:
-                return 3
-            elif land_cover == 4:
-                return 4
-            elif land_cover == 5:
-                return 3
-            elif land_cover == 6:
-                return 2
-            elif land_cover == 7:
-                return 5
-            elif land_cover == 8:
-                return 2
-            elif land_cover == 9:
-                return 5
-            elif land_cover == 10:
-                return 5
-            elif land_cover == 11:
-                return 1
-            elif land_cover == 12:
-                return 3
-            elif land_cover == 13:
-                return 3
-            elif land_cover == 14:
-                return 5
-            elif land_cover == 15:
-                return 3
-            elif land_cover == 16:
-                return 5
-
-        def classify_elevation(row):
-            elevation = row[SET_ELEVATION]
-            if elevation <= 500:
-                return 5
-            elif elevation <= 1000:
-                return 4
-            elif elevation <= 2000:
-                return 3
-            elif elevation <= 3000:
-                return 2
-            else:
-                return 1
-
-        def classify_slope(row):
-            slope = row[SET_SLOPE]
-            if slope <= 10:
-                return 5
-            elif slope <= 20:
-                return 4
-            elif slope <= 30:
-                return 3
-            elif slope <= 40:
-                return 2
-            else:
-                return 1
-
-        def set_penalty(row):
-            classification = row[SET_COMBINED_CLASSIFICATION]
-            return 1 + (exp(0.85 * abs(1 - classification)) - 1) / 100
+        penalty=GridPenalty()
 
         logging.info('Classify road dist')
-        self.df[SET_ROAD_DIST_CLASSIFIED] = self.df.apply(classify_road_dist, axis=1)
+        
+        #define bins as -inf to 5, 5 to 10, 10 to 25, 25 to 50, 50 to inf
+        bins = [float("-inf"),5,10,25,50,float("inf")]
+        
+        #define classifiers
+        classifiers = [5,4,3,2,1]
+
+        self.df[SET_ROAD_DIST_CLASSIFIED] = pd.cut(self.df[SET_ROAD_DIST], bins, labels=classifiers).astype(float)
```

Comment:
@oluchee This looks a lot neater! Can you do the same for the remaining classification steps?   I suggest naming `bins` to `road_distance_bins` and `classifiers` to `road_distance_labels`.